work16/index.js

```diff
@@ -0,0 +1,248 @@
+
+const REQUEST_URL = "https://mocki.io/v1/21ee587f-5a59-40ff-98c6-46b2d4715f25"
+
+// 間違っているURLの場合↓
+// const REQUEST_URL = "https://myjson.dit.upm.es/api/bins/ほげほげajy3";
+// 空の配列の場合↓
+// const REQUEST_URL = "https://mocki.io/v1/242b685f-a3d7-45a8-aeca-0376bd495b89";
+// 503エラーの場合↓
+// const REQUEST_URL = "https://httpstat.us/503";
+
+const request = async () => {
+    const response = await fetch(REQUEST_URL, {
+        headers: {
+            "Content-Type": "application/json"
+        }
+    });
+    if (!response.ok) {
+        console.log(`${response.status}:${response.statusText}`);
+    } else {
+        return response.json();
+    }
+}
+
+const init = async () => {
+    renderLoadingImg();
+
+    let responseJsonData;
+
+    try {
+        responseJsonData = await request();
+
+        if (!responseJsonData) {
+            return;
+        }
+    } catch (error) {
+        displayErrorMessage(error.message);
+    }
+    finally {
+        hideLoadingImg();
+    }
+
+    if (!responseJsonData.data.length) {
+        displayErrorMessage("data is empty");
+        return;
+    }
+    displayNews(responseJsonData.data);
+}
+
+const displayErrorMessage = (error) => {
+    const p = document.createElement("p");
+    p.textContent = error;
+    document.body.prepend(p);
+}
+
+const displayNews = (newsDataArray) => {
+    renderNewsTab(newsDataArray);
+    renderNewsContent(newsDataArray);
+}
+
+const renderNewsTab = (newsDataArray) => {
+    const tabList = document.getElementById("js_tabList");
+    const fragment = document.createDocumentFragment();
+
+    newsDataArray.forEach((newsCategoryObj, newsCategoryIndex) => {
+        const li = document.createElement("li");
+        const a = document.createElement("a");
+
+        if (!newsCategoryIndex) {
+            li.setAttribute("aria-selected", true);
+        } else {
+            li.setAttribute("aria-selected", false);
+        }
+        li.setAttribute("roll", "tab");
+        li.id = `${"js_tabTopics" + ++newsCategoryIndex}`;
+        li.classList = "tabTopics";
+
+        a.textContent = newsCategoryObj.field
+        a.href = "#"
+
+        li.appendChild(a);
+        fragment.appendChild(li);
+    });
+    tabList.appendChild(fragment);
+
+    const tabTopics = document.querySelectorAll(".tabTopics");
+    for (let i = 0; i < tabTopics.length; i++) {
+        tabTopics[i].addEventListener("click", () => {
+            if (tabTopics[i].getAttribute("aria-selected") === "false") {
+                for (let k = 0; k < tabTopics.length; k++) {
+                    tabTopics[k].setAttribute("aria-selected", false);
+                }
+                tabTopics[i].setAttribute("aria-selected", true);
+            }
+        });
```

Comment:
IMO
I think it is better to make it a separate function since it does more than just a function name.
Also, how about switching tabs as follows?
```suggestion
   const tabTopics = document.querySelectorAll(".tabTopics");
   for (let i = 0; i < tabTopics.length; i++) {
    tabTopics[i].addEventListener("click", (event) => {
      const selectedTab = document.querySelector('[aria-selected="true"]');
      selectedTab.ariaSelected = false;
      event.currentTarget.ariaSelected = true;
    });
  }
```

Reply:
I had not thought of this logic. It's amazing...
I wanted to do something about the nesting structure, but I was able to get around it by just using event.currentTarget and getting the element when "aria-selected = true".
Thank you so much for your help.

commit: [21e682f](https://github.com/yasu-pro/js-handson/pull/18/commits/21e682f7b0dc2929ff679fc857b8e0f8b428f3fc)